index.js

```diff
@@ -1,153 +1,157 @@
+            objects: 'always-multiline',
+            imports: 'always-multiline',
+            exports: 'always-multiline',
+            functions: 'never',
         }],
-        "arrow-spacing": "error",
-        "curly": "error",
-        "eqeqeq": "error",
-        "capitalized-comments": "error",
-        "no-unused-expressions": "error",
-        "no-useless-call": "error",
-        "no-useless-concat": "error",
-        "no-useless-escape": "error",
-        "no-useless-return": "error",
-        "vars-on-top": "error",
-        "no-undef-init": "error",
-        "no-undefined": "error",
-        "no-use-before-define": "error",
-        "callback-return": "error",
-        "no-mixed-requires": "error",
-        "array-bracket-spacing": "error",
-        "block-spacing": "error",
-        "brace-style": "error",
-        "camelcase": "error",
-        "comma-spacing": "error",
-        "comma-style": "error",
-        "computed-property-spacing": "error",
-        "eol-last": "error",
-        "func-call-spacing": "error",
-        "func-name-matching": "error",
-        "key-spacing": "error",
-        "keyword-spacing": "error",
-        "lines-around-directive": "error",
-        "new-cap": [
-            "error",
+        'padding-line-between-statements': [
+            'error',
+            // The following two lines replace the deprecated "lines-around-directive" rule.
             {
-                "capIsNewExceptions": ["$.Deferred", "jQuery.Deferred"]
+                blankLine: 'always',
+                prev: 'directive',
+                next: '*',
+            },
+            {
+                blankLine: 'any',
+                prev: 'directive',
+                next: 'directive',
+            },
+            // Following line replaces the deprecated "newline-before-return" rule.
+            {
+                blankLine: 'always',
+                prev: '*',
+                next: 'return',
             }
         ],
-        "new-parens": "error",
-        "newline-before-return": "error",
-        "no-lonely-if": "error",
-        "no-nested-ternary": "error",
-        "no-trailing-spaces": "error",
-        "no-underscore-dangle": "error",
-        "no-unneeded-ternary": "error",
-        "no-whitespace-before-property": "error",
-        "space-before-blocks": "error",
-        "space-in-parens": "error",
-        "space-infix-ops": "error",
-        "space-unary-ops": "error",
-        "spaced-comment": "error",
-        "no-compare-neg-zero": "error",
-        "no-negated-condition": "warn",
-        "no-warning-comments": "warn",
-        "global-require": "warn",
-        "require-jsdoc": "warn"
+        'arrow-spacing': 'error',
+        'curly': 'error',
+        'eqeqeq': 'error',
+        'capitalized-comments': 'error',
+        'no-unused-expressions': 'error',
+        'no-useless-call': 'error',
+        'no-useless-concat': 'error',
+        'no-useless-return': 'error',
+        'vars-on-top': 'error',
```

Comment:
I'm not sure if we want to enforce moving all variable declarations to the top.

Reply:
@alexander-schranz @cbl6499 What do you think? It's totally fine for me to remove this rule - as sometimes, e.g. after an early return, variable declarations are fine for me.

Reply:
Also fine to remove this rule.